app/domain/operations/hbx_enrollments/expire.rb

```diff
@@ -4,38 +4,193 @@ module Operations
+          msg = "No Transmittable::Transmission found with given global ID: #{transmission_gid}"
+          Failure(msg)
+        end
       end
 
-      def expire_enrollment(enrollment)
-        return Failure("Failed to expire enrollment hbx id #{enrollment.hbx_id} - #{enrollment.kind} is not a valid IVL enrollment kind") unless enrollment.is_ivl_by_kind?
+      def find_and_update_request_transaction(transaction_gid, job, request_transmission)
+        transaction = find_transaction_by_global_id(transaction_gid)
 
-        enrollment.expire_coverage!
+        if transaction.success?
+          status_result = update_status("Transmittable::Transaction found with given global ID: #{transaction_gid}",
+                                        :succeeded,
+                                        { transaction: transaction.value! })
+          return status_result if status_result.failure?
+          Success(transaction.value!)
+        else
+          add_errors(:invalid_transaction_on_enrollment_expire,
+                     "No Transmittable::Transaction found with given global ID: #{transaction_gid}",
+                     { transmission: request_transmission, job: job })
+          status_result = update_status("No Transmittable::Transaction found with given global ID: #{transaction_gid}",
+                                        :failed,
+                                        { transmission: request_transmission, job: job })
+          return status_result if status_result.failure?
+          msg = "No Transmittable::Transaction found with given global ID: #{transaction_gid}"
+          Failure(msg)
+        end
+      end
+
+      def construct_response_transmission_params(job)
+        Success({
+                  job: job,
+                  key: :hbx_enrollment_expiration_response,
+                  title: "Transmission response to expire enrollment with hbx id: #{hbx_enrollment.hbx_id}.",
+                  description: "Transmission response to expire enrollment with hbx id: #{hbx_enrollment.hbx_id}.",
+                  publish_on: Date.today,
+                  started_at: DateTime.now,
+                  event: 'received',
+                  state_key: :received,
+                  transmission_id: hbx_enrollment.hbx_id,
+                  correlation_id: hbx_enrollment.hbx_id
+                })
+      end
+
+      def construct_response_transaction_params
+        Success({
+                  transmission: response_transmission,
+                  subject: hbx_enrollment,
+                  key: :hbx_enrollment_expiration_response,
+                  title: "Enrollment expiration response transaction for #{hbx_enrollment.hbx_id}.",
+                  description: "Transaction response to expire enrollment with hbx id: #{hbx_enrollment.hbx_id}.",
+                  publish_on: Date.today,
+                  started_at: DateTime.now,
+                  event: 'received',
+                  transaction_id: hbx_enrollment.hbx_id,
```

Comment:
I think this needs to be `correlation_id`, even though the field on the transaction model is `transaction_id`.  Check out: https://github.com/ideacrew/enroll/blob/8ce54c1e297e31903338b2dfb90f77fd3e58371d/app/domain/operations/transmittable/create_transaction.rb#L39